lib/mps_simulator.h

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MPS_SIMULATOR_H_
+#define MPS_SIMULATOR_H_
+
+// For templates will take care of parallelization.
+#define EIGEN_DONT_PARALLELIZE 1
+
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <memory>
+#include <vector>
+
+#include "../eigen/Eigen/Dense"
+#include "mps_statespace.h"
+
+namespace qsim {
+
+namespace mps {
+
+/**
+ * Quantum circuit simulator without vectorization.
+ */
+template <typename For, typename fp_type = float>
+class MPSSimulator final {
+ public:
+  using MPSStateSpace_ = MPSStateSpace<For, fp_type>;
```

Comment:
You probably can just write `using MPSStateSpace = mps::MPSStateSpace<For, fp_type>;`. So `MPSStateSpace` can be without trailing underscore here.

Reply:
This gave me an error saying "Cannot re-alias": 
```
 error: declaration of 'using MPSStateSpace = class qsim::mps::MPSStateSpace<For, fp_type>' changes meaning of 'MPSStateSpace' [-fpermissive]
```